### PR TITLE
Added monitoring for testnet

### PR DESCRIPTION
**Pull request by semeniak97mf**
Actions: https://github.com/Atleta-network/atleta-network-deploy/actions/runs/10593172785
Dashboards: http://185.190.140.207:3000/dashboards

### CODE REVIEW

File: .github/workflows/deploy-testnet.yml [hunk 1]
```diff
@@ -265,6 +271,10 @@ jobs:
         scp testnet/package-lock.json ${{ secrets[matrix.user_key] }}@${{ secrets[matrix.host_key] }}:/home/${{ secrets[matrix.user_key] }}/atleta/package-lock.json
         scp config.env ${{ secrets[matrix.user_key] }}@${{ secrets[matrix.host_key] }}:/home/${{ secrets[matrix.user_key] }}/atleta/config.env
         scp chainspecs/chain-spec.testnet.json ${{ secrets[matrix.user_key] }}@${{ secrets[matrix.host_key] }}:/home/${{ secrets[matrix.user_key] }}/atleta/chainspec.json
+        ssh -o StrictHostKeyChecking=no ${{ secrets[matrix.user_key] }}@${{ secrets[matrix.host_key] }} "mkdir -p /home/${{ secrets[matrix.user_key] }}/atleta/promtail"
```
[the remark this answers is not in the document]
Comment (semeniak97mf) [reply .github/workflows/deploy-testnet.yml:274]: It is undesirable if the default user is root on one of the hosts.

File: testnet/run-validator.sh [hunk 2]
```diff
@@ -33,25 +36,56 @@ check_chainspec() {
...
     fi
+
+    if [ "$(docker ps -q -f name=$container_node_exporter)" ]; then
+        echo "Stopping existing container..."
+        docker stop "$container_node_exporter"
+    fi
+
+    if [ "$(docker ps -aq -f name=$container_node_exporter)" ]; then
+        echo "Removing existing container..."
+        docker rm "$container_node_exporter"
```
[the remark this answers is not in the document]
Comment (semeniak97mf) [reply testnet/run-validator.sh:56]: I completely agree

File: .github/workflows/deploy-telemetry-devnet.yml [hunk 3]
```diff
@@ -38,8 +64,13 @@ jobs:
       run: |
         scp utils/telemetry/docker-compose.yml ${{ secrets.DEVNET_USERNAME }}@${{ secrets.DEVNET_HOST }}:/sportchain/telemetry/docker-compose.yml
         scp utils/telemetry/docker_conf.env ${{ secrets.DEVNET_USERNAME }}@${{ secrets.DEVNET_HOST }}:/sportchain/telemetry/docker_conf.env
-        scp utils/telemetry/prometheus/prometheus.fill.yml ${{ secrets.DEVNET_USERNAME }}@${{ secrets.DEVNET_HOST }}:/sportchain/telemetry/prometheus/prometheus.yml
-        
+        ssh -o StrictHostKeyChecking=no ${{ secrets.DEVNET_USERNAME }}@${{ secrets.DEVNET_HOST }} "mkdir -p /sportchain/telemetry/prometheus"
```
[the remark this answers is not in the document]
Comment (semeniak97mf) [reply .github/workflows/deploy-telemetry-devnet.yml:67]: Of course, it's easier to read this way. Thanks

File: testnet/run-validator.sh [hunk 4]
```diff
@@ -33,25 +36,56 @@ check_chainspec() {
...
+    fi
+
+    if [ "$(docker ps -q -f name=$container_process_exporter)" ]; then
```
[the remark this answers is not in the document]
Comment (semeniak97mf) [reply testnet/run-validator.sh:59]: Bash has 'foreach' loop? It is present in zsh but not in bash. Bash has only 'for'. I'll do it with 'for'.

File: .github/workflows/deploy-telemetry-devnet.yml [hunk 5]
```diff
@@ -23,11 +24,36 @@ jobs:
         chmod 600 ~/.ssh/id_rsa
         eval $(ssh-agent)
 
-    - name: Substitute environment variables
+    - name: Substitute environment variables
```
[the remark this answers is not in the document]
Comment (semeniak97mf) [reply .github/workflows/deploy-telemetry-devnet.yml]: Done

File: testnet/run-validator.sh [hunk 6]
```diff
@@ -74,6 +109,41 @@ start_node() {
         --rpc-max-connections 10000
 }
 
+start_node_exporter() {
+
+echo "Starting the node_exporter..."
+    docker pull prom/node-exporter:latest
```
[the remark this answers is not in the document]
Comment (semeniak97mf) [reply testnet/run-validator.sh:86]: Done

File: testnet/run-validator.sh [hunk 7]
```diff
@@ -74,6 +80,43 @@ start_node() {
...
+    docker run -d --name "$container_node_exporter" \
+        -p 9100:9100 \
+        prom/node-exporter:latest
+}
+
+start_process_exporter() {
+
+    echo "Starting the process_exporter..."
+    root=$(dirname "$(readlink -f "$0")")
```
Comment (hrls) [inline testnet/run-validator.sh:95]: Define this var at script top level

File: testnet/run-validator.sh [hunk 8]
```diff
@@ -74,6 +80,43 @@ start_node() {
...
+        --procfs=/host/proc
+}
+
+start_promtail() {
+
+    echo "Starting the promtail..."
+    docker pull grafana/promtail:latest
+    docker run -d --name "$container_promtail" \
+         -p 9080:9080 \
+         -v "$(pwd)/promtail/promtail-config.yml:/etc/config/promtail-config.yml" \
```
Comment (hrls) [inline testnet/run-validator.sh:113]: Re-use `${root}` variable from the func above.

**Review by hrls — DISMISSED** [review]

File: testnet/run-validator.sh (hunk 7 above)
Comment (semeniak97mf) [reply testnet/run-validator.sh]: Done

File: testnet/run-validator.sh (hunk 8 above)
Comment (semeniak97mf) [reply testnet/run-validator.sh]: Done

**Review by hrls — APPROVED** [review]

**Review by ales-tsurko — APPROVED** [review]

**Review by DmitriVT — APPROVED** [review]

**Pull request merged**